File: backend/src/main/java/no/strazdins/rebus/dto/AnswerDto.java
package no.strazdins.rebus.dto;

import io.swagger.v3.oas.annotations.media.Schema;

/**
 * Answer for one specific challenge (one specific team).
 */
@Schema(description = "Answer for one specific challenge (one specific team)")
public record AnswerDto(
    @Schema(description = "ID of the challenge", example = "123", required = true)
    int challengeId,
    @Schema(description = "The answer text", example = "Hello World")
    String answer,
    @Schema(description = "Score given by the grader", example = "10")
    Integer score,
    @Schema(
        description = "URL of the image submitted as an answer to a challenge by a team",
        nullable = true
    )
    String imageUrl
) {
}

File: backend-v2/src/common/types/dto/answerDto.ts
/**
 * Answer DTO.
 *
 * @property challengeId ID of the challenge
 * @property answer Answer to the challenge
 * @property score Score of the answer given by the administrator
 * @property imageUrl URL of the image posted by the user
 */
export type AnswerDto = {
  challengeId: number;
  answer: string | null;
  score: number | null;
  imageUrl: string | null;
};
